### PR TITLE
Sprint 5 SonarQube Refactoring

### LINKED ISSUE #124: SonarQube refactoring for Sprint 5
This issue is based on the SonarQube code quality metrics report ran and given at the end of sprint 5. The different issues need to be addressed to have code quality fitting the acceptance criteria defined below. The code coverage metric is given to us by Clover and a report is also generated at the end of sprint 5. The reports will be linked to this issue in the comments.

Acceptance Criteria:
- [x]  Bug tolerance: 0
- [x]  Security tolerance: 0
- [x]  Code smell tolerance: 1-2
- [x]  Code dupplication tolerance: 3-4%
- ~~[ ]  Code Coverage tolerance: 80%+~~ The code coverage will be part of another issue since it makes more sense to do so. The SonarQube refactoring is already a big piece on its own. Furthermore, the release documentation separates the code coverage/testing from the quality metrics, so it makes more sense to separate them.

~~The code coverage tolerance should be a mix of line, method, and function coverage.
Clover code coverage information:
https://notorious-eng.atlassian.net/wiki/spaces/DOCUMENTAT/pages/295141377/Unit+testing~~

~~There are some unit tests that were made in the beginning but do not apply anymore (most of them)to the current version of the project. Those can be found on the 101 branch. They can serve as a basis for constructing unit tests. You can consult the branch, but do not try to update it or merge it. The branch is full of errors. Once this issue is closed, we will close the branch as well.~~

### PR DESCRIPTION
**Pull request by JayChouteau**
This pull request includes all refactoring done based on the sonarQube analysis on the master branch at the end of sprint 5. It includes bugs, vulnerabilities, code smells, and dupplication. Most of them were fixed. A few of them were resolved as won't fix since the code it was located in was currently being refactored or modified in different issues. The acceptance criteria of the linked issue were fulfilled. 

This pull request should be merged after (on top) of all the current pull-requests so all the little refactoring changes apply on top of the develop branch. All information pertinent to the SonarQube is available on Google Drive in quality metrics. The description of issues and resolution of those will have to show in the release 2 documentation as well.

### CODE REVIEW

File: Server/src/main/java/local/DatabaseConnection.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
     //Make sure to set the user and password to the proper values.
     //Credentials should be set to that which you are using on your local DB server.
     private static final String USER = "root"; // todo  use username and password specific to your machine
-    private static final String PASSWORD = "";
+    private static final String PASSWORD = "root";
```
Comment (jeffbuonamici) [inline Server/src/main/java/local/DatabaseConnection.java]: Just remove this.

**Review by jeffbuonamici — CHANGES_REQUESTED** [review]
Thanks for the code clean up. There's just one small change.

File: Server/src/main/java/local/DatabaseConnection.java (hunk 1 above)
Comment (JayChouteau) [reply Server/src/main/java/local/DatabaseConnection.java]: Hey Jeff, it is now resolved!

**Comment by JayChouteau** [comment]
In the last commit, I removed some methods and files that were unused in our application. I left the placeholder for the feature to come (threshold and recommendation) and didn't remove anything from the controller part and data-preprocessing since there was a lot of work that was done on those that are not yet merged into our develop branch.

**Comment by PaulMicu** [comment]
can you please put a screenshot of sonarqube and clover after your changes

**Review by PaulMicu — APPROVED** [review]
some of these changes seem silly but if that's what sonarqube wants i guess it's ok

**Review by jeffbuonamici — APPROVED** [review]
Looks good now.

**Pull request merged**